app/routes.js

```diff
@@ -40,19 +42,26 @@ export const requireAuth = (api) => (nextState, replace) => {
  *
  * @return {boolean|null} returns true if hash mapping happened
  */
-export const requireAuthAndNoPatient = (api) => (nextState, replace, cb) => {
-  if (!api.user.isAuthenticated()) {
+export const requireAuthAndNoPatient = (api, store) => (nextState, replace, cb) => {
+  let { blip: state } = store.getState();
+
+  if (!state.isLoggedIn && !api.user.isAuthenticated()) {
     replace('/login');
     return cb();
-  }
-  else {
-    api.user.get(function(err, user) {
-      if (personUtils.isPatient(user)) {
+  } else {
+    if (state.loggedInUser) {
+      if (personUtils.isPatient(state.loggedInUser)) {
```

Comment:
Is `state.loggedInUser` currently a full user object? In order for this to work, I think it must be, but eventually in the normalized state we'll want `loggedInUser` to be just a `userid`. So I imagine in the future this will have to be changed to something like `if (personUtils.isPatient(state.allUsers[state.loggedInUser])`...